### ISSUE TITLE
Gunung Agung Animation Design

### ISSUE THREAD

**Issue by awright** [hotosm/visualize-change#5]
This is a prototype implementation of visualization tool that can be applied specifically to the Gunung Agung geographic area. This extends the processing work of #3 with a specific emphasis on the user interface. This version of the animation toolkit as demonstrated by the Gunung Agung use case, will include: 

- [x] #7 Initial implementation of frontend UI + server API (HTTP)
- [ ] #6 Alpha version of the rendering API

**Comment by szymonkaliski** [comment]
The current state of the tool already allows for rendering of specified lat/lng (zoom level is currently hardcoded at level 12).

I'd be happy to work on renderer keeping in mind what you need for those animations. Is there specific style, set of features, etc... that you'd like to see?

Also, this has deadline of Feb 16th, right? That deadline is in regards to generated MP4 I hope? :) We wont have polished frontend before end of February.

**Comment by awright** [comment]
@smit1678 any comment on specific style or features for the Gunung Agung Animation  cc @mataharimhairi 

(Yes @szymonkaliski - this is for the Feb 16th deadline!)

**Comment by awright** [comment]
Timeframe: the alert was increased to the highest level at the volcano on September 22, so the timeframe could focus on the mapping progress for the island of Bali **from August through to December 2017 or up to February 2018**, when the animation is produced.

**Comment by szymonkaliski** [comment]
Per our meeting today, to implement Frontend Alpha version we would like some comments / "green light" on the Wireframes here (https://github.com/hotosm/visualize-change/issues/2#issuecomment-361926613).

**Comment by mataharimhairi** [comment]
@awright here is the Tasking Manager project that was set up for mapping the area around Gunung Agung in response to the heightened volcanic activity on the 22nd September 2017:

http://tasks.openstreetmap.id/project/221

The focus of the task was to map road networks (primary and secondary roads) as well as buildings, so we could visualise all the roads and buildings. Style wise, perhaps we can use the Humanitarian style?

**Comment by hoffmannjan** [comment]
We'll have styles implemented and render tested with them at the end of next week, so we'll post you the effects then.

**Comment by smit1678** [comment]
> Style wise, perhaps we can use the Humanitarian style?

@mataharimhairi @awright @hoffmannjan what do we think is the goal of a style we choose? If we go with a very verbose style like the humanitarian style I feel like we'll lose the ability to bring out the changes. But if we want to see what the map looks like as it grows, that's different. Or if we go with a simpler style with more contrast between the data that is changing and the background, then the changes show up more.

**Comment by mataharimhairi** [comment]
@smit1678 I think you're right. Perhaps something along the lines of the style used by Mapbox to visualise the malaria mapping would have more of an impact when presenting the change in edits

**Comment by hoffmannjan** [comment]
OK, so we'll experiment with few options to select so you can play with them and give us feedback.

**Comment by awright** [comment]
@mataharimhairi to confirm - this is the Mapbox tool and style you are referring to? https://www.mapbox.com/bites/00343/#5.8/-19.036/24.323 cc @hoffmannjan

**Comment by hoffmannjan** [comment]
https://www.dropbox.com/s/kovumip1gwbuq15/render-01.mp4?dl=0
& waiting for your feedback :).

**Comment by awright** [comment]
let's talk about it in 2 ;) #21 but it would be great to see a couple of versions. but let's give solid feedback first on this one. thanks!

**Comment by awright** [comment]
Mapbox and ITO have been design inspirations: 
- https://www.youtube.com/watch?v=7sC83j6vzjo
- https://www.mapbox.com/bites/00343/#5.8/-19.833/25.875

**Comment by awright** [comment]
testing what is technically possible: 
- highlight on what is happening on given data
- go into lower timeframes

**Comment by hoffmannjan** [comment]
**lat:** -8,417705731174067
**lng:** 115,6061274949788
**interval:** day
**date span:** 01-09-2017 - 30-10-2017

[video](https://www.dropbox.com/s/hvu1sclx6ntwybk/render-01-daily.mp4?dl=0)

**lat:** -8,417705731174067
**lng:** 115,6061274949788
**interval:** hour
**date span:** 03-10-2017 - 08-10-2017

[video](https://www.dropbox.com/s/qvhl3tfdeswdpb3/render-01-hourly.mp4?dl=0)

So as you can see we added highlight for features added in latest date span. In rendered mp4 It works and look well, but in preview it make whole animation sluggish / frames dropping.
Also, played with hour interval and It makes progress more visible, but still main issue is in data.

**Comment by szymonkaliski** [comment]
**I think we are ready to have the final render generated by Friday, if you give us the final options that you want to use.**

We can also put this on the test server (with very crude UI) so you can play with it and generate your own renders, but keep in mind that on the test server there's on underzooms, and for the renders we're doing on our own laptops we can generate the underzoom just for Indonesia.

**Comment by awright** [comment]
HOT to provide feedback on:
- Right place? Right dates?
- Map Style 
cc @mataharimhairi @smit1678

**Comment by awright** [comment]
In terms of where and when:

The Gunung Agung animation is being created for dfat.gov.au/. An alert was increased to the highest level at the volcano on September 22, so the timeframe should focus on the mapping progress for the island of Bali from **August through to December 2017** cc  @szymonkaliski

**Comment by awright** [comment]
@mataharimhairi do you want the geographic extent of the animation to zoom in or out? to perhaps move around? any guidance is welcome. ;)

**Comment by awright** [comment]
@szymonkaliski we will provide you design feedback on your styles for the animation. you have the date period. lets stick to the same geographic extent for the time being.

**Comment by szymonkaliski** [comment]
>lets stick to the same geographic extent for the time being.

Not sure if I understand, should we stick to the lat/lng that we used for test renders so far? (lat: -8,417705731174067 lng: 115,6061274949788)

**Comment by awright** [comment]
Yes.

**Comment by mataharimhairi** [comment]
@awright @szymonkaliski sorry for the late response. ideally for the visualisation, it would be great if it could start being zoomed out to the world view, spinning to Indonesia, and then Bali. This would provide the user with more geospatial context as to where Gunung Agung is.

**Comment by szymonkaliski** [comment]
@mataharimhairi while I agree this would look super nice, we don't have the technology to go from 3d to 2d view, or animate zoom at the moment.

We could add zoom animations, but this would mean that we need to generate the mp4/gifs with 60 frames per second (60 images per each second of the animation), instead of current six images per frame (we just keep one image for few seconds on the screen and then move to the next), this is doable, but will make the generation process at least ten times as long as it is right now, so this is something important to keep in mind for the server setup -- the generation processes are scheduled one after another, so if it takes few hours for each animation, I'm afraid we can easily hit situation where someone will have to wait few days or longer for their animation.

So, to recap, we need to know about the constraints we have:

1. 3d globe is not possible with this technology we are using
2. "smooth zooming" animation (mp4) is possible (would "just" have to be implemented), but requires a lot more time (so a lot more CPU usage and disk usage) for generation of the rendered mp4/gifs

Also important to keep in mind that we only support zoom levels 12+, so if we zoom from level 0, we will only see the geographical data, before we can see roads and buildings on higher zoom levels.

Let me know what you think, and if you think spending time on adding this "zooming" step is worth it.

**Comment by mataharimhairi** [comment]
@szymonkaliski thanks for the quick response and for explaining the technical constraints. that's fine that 3d globe is not possible, and i now understand that the processing load will be much larger (x10) if we were to implement zoom animations.

Would it be possible to include the zoom animation just for the Gunung Agung visualisation, as this is a key deliverable for one of our project agreements? It is not a priority that this feature is available for all users and their visualisations created.

It is also fine that we only see geographical data, and not the map features such as buildings and roads when at level 0, as it is the geographical data that will provide the viewer with context as to where the area of interest is on the globe!

Hope that makes sense.

**Comment by szymonkaliski** [comment]
@mataharimhairi sure, we can implement the zoom animation only for Gunung Agung, but that would mean that:

1. we will do some (non-trivial) work that we will have to throw out
2. it's error-prone to support two different versions of the application, so we would have to first finalise the renders of Gunung Agung, before moving back to doing any work in the "main" application - otherwise I'm afraid of strange bugs popping up, due to different implementations of the video renderer, depending on if we work for main application, or this specific animation export

I'd also like to ask, if there's anything else that you'd like to have changed in this animation - if there's anything that we want to work/look differently (besides colors and background layer - these are configurable), and also have it "stay" for the main application, then I'd prefer to implement this first, and then add the zoom animation just for this. This is just because then it should be bit easier to "undo" the zooming animation and still keep the other features intact.

Let me know if there's anything I'm not explaining too clearly :) I'm happy to do the work, I'd just like to have a confirmation that the approach I'm describing is understood, and we should move forward with it.

**Comment by mataharimhairi** [comment]
@szymonkaliski It makes a whole lot more sense when you put it that way. We can chat about it during tomorrow's meeting, but I would agree that there is no point adding more code to the tool that won't be used in the end product. Especially as time is limited and it can lead to errors in the final code.

Are the current versions the videos hosted on Dropbox that were posted on this issue by Jan, 13 days ago? Could you please send/attach the newest versions so that I can provide some current feedback. Thank you!

**Comment by szymonkaliski** [comment]
>Are the current versions the videos hosted on Dropbox that were posted on this issue by Jan, 13 days ago?

Yes, these are the latest versions of test renders we did to show what's possible. 
What we can configure:

- lat/lng and zoom level - we can set any zoom level for this animation since we generated the underzooms just for Indonesia
- color and width of roads (or turn off)
- color and width of roads "highlight" (or turn off)
- color and width of buildings (or turn off)
- color and width of buildings "highlight" (or turn off)
- background layer (anything from mapbox)

**Comment by mataharimhairi** [comment]
@szymonkaliski Could you please make the following changes to the animation:

- it is much too zoomed in at the moment. it would be more effective to be zoomed to the extent where you can see the whole island of bali. this will show the impact of the mapping on the eastern side of the island in response to the heightened volcanic activity
- i would leave all the colours, width and highlights of the buildings as it is
- i would also leave the highlights of the roads, but could we make the colour more opaque and change the width to something finer. hopefully this will make the road features more defined
- currently the final features, after the highlight look a little out of focus 
- similar to the malaria mapbox example, can we make the place names stand out a little more. perhaps this can be done by making the colour whiter or the font bold. the names should also be above the building and road features

**Comment by hoffmannjan** [comment]
hi, new renders based on @mataharimhairi suggestion.
I changed layers order so the names are always standing at the top, hope this makes them easy to read .
With roads feature width, we still use 1px - this becomes very unreadable at this level of zoom with bigger width.

**lat**: -8,47405695155274
**lng**: 115,20304774811518
**interval**: day
**date span**: 01-09-2017 - 30-10-2017

[video](https://www.dropbox.com/s/z9a7aryyerbtng9/render-02-daily.mp4?dl=0)

**lat:** -8,47405695155274
**lng:** 115,20304774811518
**interval**: hour
**date span**: 03-10-2017 - 08-10-2017

[video](https://www.dropbox.com/s/z9a7aryyerbtng9/render-02-hourly.mp4?dl=0)

**lat:** -8,47405695155274
**lng:** 115,20304774811518
**interval**: hour
**date span**: 01-01-2017 - 01-12-2017

[video](https://www.dropbox.com/s/rh61f73v0cfch1s/render-02-weekly.mp4?dl=0)

**Comment by awright** [comment]
This is looking nicer and nicer. I imagine @mataharimhairi will have some
more feedback, but looking good for sure!

**Comment by mataharimhairi** [comment]
@hoffmannjan This looks great! 

I particularly like the last video, where it shows the mapping for the whole year, at weekly intervals. Showing the lead up of mapping in the area of interest, before the actual focus dates helps to demonstrate the impact of the heightened volcano activity on the mapping efforts of the eastern side of Bali.

Is it possible to slow down the video? Currently it plays very fast, which makes it difficult to take in the changes. It would also be great if the dates for the timeframes are displayed.

I would like to share a video with the donors for some of their feedback, and I think we are getting to a good point at which we can do that. Could you please create a video from the 1st September 2017 to the 1st March 2018, at a daily interval, slowed down, with the dates displayed?

Thank you

**Comment by hoffmannjan** [comment]
Hi,

here is new video:

**lat:** -8,47405695155274
**lng:** 115,20304774811518
**interval:** week
**date span:** 01-03-2017 - 01-03-2018

[video](https://www.dropbox.com/s/vrvjiwnhbrni6tz/render-03-daily.mp4?dl=0)

As you requested, I changed frames per seconds from 10 to 7, so video is a little bit slower.
Also, we believe that putting dates / text on video files can be bad decision - it easily can be unreadable and not clear, but if you think It's necessary - we can play with it.

Looking forward for you feedback :)

**Comment by awright** [comment]
@hoffmannjan thanks jan. can we see a version with dates / text as @mataharimhairi suggests? it is what dates and time interval we are seeing in the animation. for two points of confusion - 
- the title of the video is render-03-daily.mp4 but you note the interval is weekly. 
- what year are we even talking about. 

perhaps adding some contextual test is something we can user test with donors.

**Comment by awright** [comment]
actually @hoffmannjan @szymonkaliski let's focus on #27 the UI workflow as that is highest priority and the download experience is second. i think with your current revisions we have something to get initial feedback with 🥂 yet the interaction of the UI will really influence the experience and impact of creating such animations.

**Comment by mataharimhairi** [comment]
@hoffmannjan @szymonkaliski 

This version of the Gunung Agung animation is fantastic! It really shows the impact of the volcanic activity on mapping the eastern side of the island. You can distinctly see the mapping wave periods in association with the volcanic alert levels. Thank you very much for slowing down the frames per seconds, I think this really helped. 

We do need to provide the dates associated with the time frames of the downloaded video, to provide context to the viewer/downloader. Perhaps this doesn't have to be on the video, maybe a text file generated with some information would suffice? 

Let's chat/brainstorm during the weekly meeting. Thank you once again. Really stoked with these results!!!

**Comment by awright** [comment]
03-07 Meeting Update: 
- Looking really good. Animation shows good lead up to editing activity. 
- Need to have reference to the dates. (Not link to website as described in view in #27 UI Design) 
- Can put information in email that you share "send". Get email with a link and description.  
- Icing (but not immediate requirement): Would like to see date @mataharimhairi. Would be similar in play bar design in slide 6 of #27 Revised UI

**Comment by awright** [comment]
- @mataharimhairi feedback on Animation - colors we choose for roads and buildings not for visual impaired and color blind. @szymonkaliski will look to get ones that are nearby for colorblind.  (also #31 )

**Comment by mataharimhairi** [comment]
Here is a site that assists with choosing tones that are more colour blind friendly: 

http://www.colouroracle.org

We should also have the colours chosen, set as default in the tool, as other users might not initially take this into account either!

**Comment by hoffmannjan** [comment]
Hi,

updated version :).

https://www.dropbox.com/s/0y059itr9ht3yoa/colorblind-supported.mp4?dl=0

**Comment by awright** [comment]
@mataharimhairi can you look over this milestone and close it. @szymonkaliski and i think this is complete but you have the clearest view of stakeholders.

**Comment by mataharimhairi** [comment]
@awright @szymonkaliski this is very close. but I don't think the requested changes (noted above) were made to the last animation? the colour selection, speed, interval and date range are perfect. the only thing is the difficultly in seeing the buildings against the road data. could we change the layering of data, so that buildings are above the roads?

**Comment by mataharimhairi** [comment]
Hello @hoffmannjan @szymonkaliski and @awright there seems to be more differences between this and the last video besides the layering order of the buildings and roads. This video is 3 mins, while the last one is only 1 min. Also, there seems to be less map features present at the end of the the most recent visualisation.

Previous Animation:
<img width="1181" alt="screen shot 2018-04-18 at 18 27 49" src="https://user-images.githubusercontent.com/8972913/38966403-efb3f850-4336-11e8-8f09-b66a2605cb20.png">

Current Animation:
<img width="1183" alt="screen shot 2018-04-18 at 18 29 36" src="https://user-images.githubusercontent.com/8972913/38966410-f9cc2862-4336-11e8-9b81-917fabd6e4cf.png">

The previous animation was perfect in regards to the dates, time frame speed and overall length of the video. The only thing that needed to be changed was the layer order of the buildings and roads.

**Comment by hoffmannjan** [comment]
Hi,

sorry for the video length difference - we were playing with FPS settings and there are some major differences in our code base since last version of this animation was posted.

But I can't see the problem with last frame that you posted: this is how the video looks at the end in QuickTime:
![screenshot 2018-04-19 09 50 33](https://user-images.githubusercontent.com/3756160/38978592-2ad102c6-43b8-11e8-8d0f-cb00716d6c83.png)

There is fixed version with same length as last time, the only difference is resolution - the last one was 1280x800 and this is 1920x1080.

[video](https://www.dropbox.com/s/s69mwygudvnlmmw/render-05.mp4?dl=0)

**Comment by mataharimhairi** [comment]
@hoffmannjan that's strange, regarding the end of the video and how the map looked. I was using VLC, so I wonder if that had something to do with it. thank you very much for the updated visualisation though! it's perfect. I'm going to close this issue to help with finalising the tickets on GitHub.
